crates/diesel_models/src/query/merchant_connector_account.rs

```diff
@@ -149,3 +159,122 @@ impl MerchantConnectorAccount {
         }
     }
 }
+
+#[cfg(all(feature = "v2", feature = "merchant_connector_account_v2"))]
+impl MerchantConnectorAccount {
+    pub async fn update(
+        self,
+        conn: &PgPooledConn,
+        merchant_connector_account: MerchantConnectorAccountUpdateInternal,
+    ) -> StorageResult<Self> {
+        match generics::generic_update_by_id::<<Self as HasTable>::Table, _, _, _>(
+            conn,
+            self.id.to_owned(),
+            merchant_connector_account,
+        )
+        .await
+        {
+            Err(error) => match error.current_context() {
+                errors::DatabaseError::NoFieldsToUpdate => Ok(self),
+                _ => Err(error),
+            },
+            result => result,
+        }
+    }
+
+    pub async fn delete_by_id(conn: &PgPooledConn, id: &str) -> StorageResult<bool> {
+        generics::generic_delete::<<Self as HasTable>::Table, _>(conn, dsl::id.eq(id.to_owned()))
+            .await
+    }
+
+    pub async fn find_by_merchant_id_connector(
+        conn: &PgPooledConn,
+        merchant_id: &common_utils::id_type::MerchantId,
+        connector_label: &str,
+    ) -> StorageResult<Self> {
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::merchant_id
+                .eq(merchant_id.to_owned())
+                .and(dsl::connector_label.eq(connector_label.to_owned())),
+        )
+        .await
+    }
+
+    pub async fn find_by_profile_id_connector_name(
+        conn: &PgPooledConn,
+        profile_id: &str,
+        connector_name: &str,
+    ) -> StorageResult<Self> {
+        generics::generic_find_one::<<Self as HasTable>::Table, _, _>(
+            conn,
+            dsl::profile_id
+                .eq(profile_id.to_owned())
+                .and(dsl::connector_name.eq(connector_name.to_owned())),
+        )
+        .await
+    }
+
+    pub async fn find_by_merchant_id_connector_name(
```

Comment:
we can deprecated this function and its usage, this related functionalities should be handled by id